### DISCUSSION TITLE
[BUG] dependent: :nullify not working on has_one relationship

### DISCUSSION THREAD

**Discussion by ppascualv** [mongodb/mongoid#5724] [General]
```
class HtmlContent
  include Mongoid::Document

  has_one :user, class_name: 'User', inverse_of: :html_content, validate: false, dependent: :nullify
end
```

```
class User
  include Mongoid::Document

  belongs_to :html_content, class_name: 'HtmlContent', inverse_of: :user, dependent: :destroy, validate: false, optional: true
end
```

```
user = User.create!
html_content = HtmlContent.create!(user: user)

html_content.destroy! # /lib/mongoid/association/depending.rb:110:in `_dependent_nullify!'
```

Raises an error:
NoMethodError: undefined method `nullify' for #<User _id: 640084c257f2fc7f67cf1862

relation.nullify
                  ^^^^^^^^
/lib/mongoid/association/depending.rb:110:in `_dependent_nullify!'

mongoid-7.5.3

**Comment by jamis** [comment]
Thank you for reporting this! I've added a ticket in our tracking system; you can follow it here if you'd like: https://jira.mongodb.org/browse/MONGOID-5685

**Comment by jamis** [comment]
I just took a look at this, and your example works for me (doesn't raise an exception) in Mongoid 7.5.4. Can you try updating to 7.5.4 and seeing if it works for you there?

**Reply by ppascualv** [reply]
Will do

**Comment by jamis** [comment]
I'm going to go ahead and close this for now; but if the issue can be duplicated with Mongoid 7.5.4, please feel free to reopen and let us know. 👍

**Reply by ppascualv** [reply]
Working on new version
